File: common-lib/data/src/main/java/com/yellowsunn/common/protocol/ResultResponse.java
package com.yellowsunn.common.protocol;

public record ResultResponse<T>(
        T data,
        boolean success,
        String message
) {
    public static <T> ResultResponse<T> ok(T data) {
        return new ResultResponse<>(data, true, null);
    }

    public static ResultResponse<Void> fail(String message) {
        return new ResultResponse<>(null, false, message);
    }
}

File: user-service/src/main/kotlin/com/yellowsunn/userservice/api/ResultResponse.kt
//package com.yellowsunn.userservice.api
//
//data class ResultResponse<T>(
//    val data: T?,
//    val success: Boolean,
//    val message: String?,
//) {
//    companion object {
//        fun <T> ok(data: T?): ResultResponse<T> =
//            ResultResponse(
//                data = data,
//                success = true,
//                message = null,
//            )
//
//        fun fail(message: String?): ResultResponse<Unit> =
//            ResultResponse(
//                data = null,
//                success = false,
//                message = message,
//            )
//    }
//}
